libs/ui/src/hooks/smartcard_auth/auth_helpers.ts

```diff
@@ -116,13 +121,127 @@ export function buildCardStorage(
  * generated by buildCardStorage
  */
 export function buildCardProgramming(
-  cardSummary: CardSummaryReady
+  cardSummary: CardSummaryReady,
+  cardApi: Card,
+  cardWriteLock: Lock,
+  logger?: Logger
 ): CardProgramming {
   return {
     programmedUser: parseUserFromCardSummary(cardSummary),
-    programUser: /* istanbul ignore next */ () => {
-      // A stub soon to be populated
-      return Promise.resolve(ok());
+    programUser: async (userData) => {
+      const { role } = userData;
+      if (logger) {
```

Comment:
Is there any way to refactor this so that the logging is separate from the logic? I often find it difficult to read code that has logging interleaved. One idea might be to put a wrapper around this function that looks at the results. Not a big deal, just a suggestion

Reply:
I was admittedly skeptical of how much of a readability difference this would make for this PR (I think separating logging made a lot of sense for https://github.com/votingworks/vxsuite/pull/2036), but I gave it a shot and agree! https://github.com/votingworks/vxsuite/pull/2140/commits/ba7441bc681055c0fb8a594379a05331632726cd